Android/app/src/main/java/app/intra/socks/ReliabilityMonitor.java

```diff
@@ -123,7 +151,66 @@ public void onError(Pipe pipe, Exception exception) {
       Bundle event = new Bundle();
       event.putInt(Names.BYTES.name(), uploadBytes);
       event.putInt(Names.CHUNKS.name(), uploadCount);
-      FirebaseAnalytics.getInstance(context).logEvent(Names.EARLY_RESET.name(), event);
+      if (uploadBuffer != null) {
+        new Thread(new Retry(context, getAddress(pipe), uploadBuffer, event)).start();
+      }
+    }
+  }
+
+  /**
+   * Class for testing whether an unreachable server becomes reachable if we reduce the size of
+   * the initial packet.
+   */
+  private static class Retry implements Runnable {
```

Comment:
Maybe name it SplitTest or something that indicates this is a side test

Reply:
Done